roles/openshift_health_checker/openshift_checks/logging/podlogs.py

```diff
@@ -0,0 +1,143 @@
+        pods, error = super(PodLogQueryTime, self).get_pods_for_component(
+            self.execute_module,
+            self.logging_namespace,
+            "kibana",
+            task_vars,
+        )
+
+        if error:
+            return {"failed": True, "changed": False, "msg": error}
+
+        kibanna_pods = super(PodLogQueryTime, self).running_pods(pods)
+        if not len(kibanna_pods):
+            msg = ('No Kibana pods were found to be in the "Running" state.'
+                   'At least one Kibana pod is required in order to perform this check.')
+            return {"failed": True, "changed": False, "msg": msg}
+
+        self.logging_namespace = get_var(task_vars, "openshift_logging_namespace", default=self.logging_namespace)
+        pods, error = super(PodLogQueryTime, self).get_pods_for_component(
+            self.execute_module,
+            self.logging_namespace,
+            "es",
+            task_vars,
+        )
+
+        if error:
+            return {"failed": True, "changed": False, "msg": error}
+
+        es_pods = super(PodLogQueryTime, self).running_pods(pods)
+        if not len(es_pods):
+            msg = ('No Elasticsearch pods were found to be in the "Running" state.'
+                   'At least one Elasticsearch pod is required in order to perform this check.')
+            return {"failed": True, "changed": False, "msg": msg}
+
+        failed = False
+        msg = ""
+
+        try:
+            uid = self.curl_kibana_with_uuid(kibanna_pods[0], task_vars)
+            self.wait_until_cmd_or_err(es_pods[0], uid, pod_log_timeout, task_vars)
+        except OpenShiftCheckException as error:
+            failed = True
+            msg = str(error)
+        except KeyError as error:
+            failed = True
+            msg = "Invalid json response from pod: {}".format(str(error))
+        except ValueError as error:
+            failed = True
+            msg = "Invalid response from pod {}".format(str(error))
+
+        return {"failed": failed, "msg": msg}
+
+    def wait_until_cmd_or_err(self, es_pod, uid, timeout_secs, task_vars):
+        """Wait a maximum of timeout_secs for the uuid logged in Kibana to be found in
+        the Elasticsearch logs. Since we are querying for a message with the the unique
+        uuid from earlier, there should only be a single match.
+        Raise an OpenShiftCheckException if the timeout is reached before a result is found."""
+
+        interval = 1  # amount of time to wait between retries
+        orig_timeout = timeout_secs
+
+        while timeout_secs > 0:
+            time.sleep(interval)
+            timeout_secs -= interval
+
+            total = self.query_es_from_es(es_pod, uid, task_vars)
+            if total == 1:
+                return
+
+        msg = "expecting match in Elasticsearch for message with uuid {}, but no matches were found after {}s."
```

Comment:
@rhcarvalho @sosiouxme sounds good, updated `curl` commands to have a `connect-timeout` of `30s` and a `max-time` of `60s`. Also, updated total time per loop iteration to take into account total `curl` time as well:https://github.com/openshift/openshift-ansible/pull/4519/commits/29149c2a6d246a1dc91e6e751f9312be74c45cfd